packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,148 @@
+import { parseAncillaryData } from "@uma/common";
+import { BN } from "../types";
+import type { Logger } from "winston";
+import { InsuredBridgeL1Client, Relay } from "../clients/InsuredBridgeL1Client";
+import { InsuredBridgeL2Client, Deposit } from "../clients/InsuredBridgeL2Client";
+
+const { toBN, toWei } = Web3.utils;
+
+enum isRelayValid {
+  No, // Should be 0
+  Yes, // Should be 1
+}
+
+interface Params {
+  logger: Logger;
+  web3: Web3;
+  l1Client: InsuredBridgeL1Client;
+  l2Client: InsuredBridgeL2Client;
+}
+
+interface RelayAncillaryData {
+  depositId: number;
+  depositTimestamp: number;
+  recipient: string;
+  l2Sender: string;
+  l1Token: string;
+  amount: number;
+  slowRelayFeePct: number;
+  instantRelayFeePct: number;
+  quoteTimestamp: number;
+  realizedLpFeePct: number;
+  slowRelayer: string;
+  depositContract: string;
+  priceRequestTime: string;
+}
+
+// Allows user to respond to a "relay" price request that was sent in response to a "deposit" on a InsuredBridge
+// deployed to an L2 network. The relay price request is submitted on L1. This pricefeed will respond "Yes" or "No"
+// based on whether the relay was correctly constructed to match a deposit. The price request includes parameters in
+// its ancillary data that must be parsed from the hex ancillary data.
+export class InsuredBridgePriceFeed extends PriceFeedInterface {
+  private readonly decimals: number;
+  private readonly logger: Logger;
+  private readonly web3: Web3;
+  private readonly l1Client: InsuredBridgeL1Client;
+  private readonly l2Client: InsuredBridgeL2Client;
+  private relays: Relay[] = [];
+  private deposits: Deposit[] = [];
+  private readonly toBNWei: (_number: number) => BN;
+
+  /**
+   * @notice Constructs the InsuredBridgePriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider to connect to Ethereum network.
+   * @param {Object} l1Client Fetches and returns latest state of L1 pool and admin contracts.
+   * @param {Object} l2Client Fetches and returns latest state of L2 deposit contract.
+   */
+  constructor({ logger, web3, l1Client, l2Client }: Params) {
+    super();
+
+    this.decimals = 18;
+    this.logger = logger;
+    this.web3 = web3;
+    this.l1Client = l1Client;
+    this.l2Client = l2Client;
+    this.toBNWei = (_number) => toBN(toWei(_number.toString()));
+  }
+
+  // This method returns the validity of a relay price request attempt. The relay request was valid if and only if it:
+  // The price request's ancillary data contains parameters that match with an L2 deposit event.
+  public async getHistoricalPrice(time: number | string, ancillaryData: string): Promise<BN> {
+    // Parse ancillary data for relay request and find deposit if possible with matching params.
+    const parsedAncillaryData: RelayAncillaryData = (parseAncillaryData(
+      ancillaryData
+    ) as unknown) as RelayAncillaryData;
+    const matchedDeposit = this.deposits.filter(
```

Comment:
Note @mrice32 and @chrismaree this method has changed. L1 relay data is no longer checked. Instead, the ancillary data is compared against L2 deposit event data. 

- If the ancillary data matches and the realized fee % is correct, then return 1.
- All other cases return 0.

This means that if someone submits a relay price request/proposal with ancillary data matching a deposit, then the OO will not dispute it, even if there no corresponding relay event. I see no risk to this.

Reply:
Are there any fields we're not checking here? If so, do you think that's a signal that they can be safely removed from the ancillary data since they aren't necessary to determine whether the request was valid or not?

Reply:
Yeah I removed `slowrelayer`